code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -648,6 +648,14 @@ var/global/list/damage_icon_parts = list()
 /mob/living/carbon/human/update_inv_wear_mask(var/update_icons=1)
 	var/obj/item/mask = get_equipped_item(slot_wear_mask_str)
 	var/obj/item/head = get_equipped_item(slot_head_str)
+	update_hair(0)	//rebuild hair
+	update_inv_ears(0)
+
+	if(!(head && (head.item_flags & ITEM_FLAG_AIRTIGHT)))
+		set_internals(null)
+	if(!(mask && (mask.item_flags & ITEM_FLAG_AIRTIGHT)))
```

Comment:
This should be `||`, you can have an airtight mask or helmet and preserve your internals.

Reply:
It might need to be changed in other places then. I think there's some head code that still does it like that.

Reply:
Well, still needs to be changed here.

Reply:
Wait, the && is being a nullguard for the flag check afterwards. It doesn't make sense to slap || in there? 

Also, if I can be honest, I really don't think checking for internals should be done in the inventory updates to begin with.. It probably should be handled via dropped() or something?

Reply:
I misspoke - what I meant was:
````
	if(!(head && (head.item_flags & ITEM_FLAG_AIRTIGHT)) && !(mask && (mask.item_flags & ITEM_FLAG_AIRTIGHT)))
		set_internals(null)
````

Reply:
Oh alright!

Reply:
Looking at this again, I think the helmet check is actually wrong. That was part of what I moved over, but it makes no sense. So gonna just leave the mask one I added.